scipy/io/matlab/streams.pyx

```diff
@@ -19,14 +19,6 @@ cdef extern from "Python.h":
         pass
     ctypedef struct FILE
 
-cdef extern from "py3k.h":
-    FILE* npy_PyFile_Dup(object file, char *mode) except NULL
-    int npy_PyFile_DupClose(object file, FILE *handle) except -1
-    int npy_PyFile_Check(object file)
-
-
-cdef bint IS_PYPY = ('__pypy__' in sys.modules)
```

Comment:
Can you clarify why removing this `pypy`-related code make sense in the Python 2 maintenance cleanup?

I'm not saying it is wrong to remove it, but we don't currently actively test `pypy` in CI (though we used to) and another reviewer may wonder the same when they are looking through this PR.

Reply:
This line should have been removed in https://github.com/scipy/scipy/pull/11655

A quick grep suggests that no uses of `IS_PYPY` in c code, in python it's defined and used in several unrelated places (and defined but not used in `scipy/io/matlab/tests/test_streams.py`)

Reply:
Yeah, I saw it used in a few places with `git grep` so wasn't clear on it

Reply:
But, strictly speaking, that' s not related to the Python 2 stuff, right?

Reply:
it as used by a Python2 only method so IDK, but it's a non issue because I moved it to the other PR.